File: HolyChildhood/Models/Page.cs
using System.Collections.Generic;

namespace HolyChildhood.Models
{
    public class Page
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int Index { get; set; }
        public int? MenuItemId { get; set; }
        public MenuItem MenuItem { get; set; }
        public List<PageContent> PageContents { get; set; }
        public Page Parent { get; set; }
        public List<Page> Children { get; set; }
    }
}

File: HolyChildhood/client/src/models/Page.ts
import {PageContent} from "./PageContent";
import {Menu} from "./Menu";

export interface Page {
    id: number;
    index: number;
    title: string;
    children: Array<Page>;
    parent: Page;
    pageContents: PageContent[];
    menuItem: Menu;
    menuItemId: number;
}
